src/rewrite_infinite_loops.cpp

```diff
@@ -0,0 +1,328 @@
+/**
+ * @file rewrite_infinite_loops.cpp
+ *
+ * @author  Alex Henrie
+ * @license GPL v2+
+ */
+
+#include "rewrite_infinite_loops.h"
+
+#include "chunk.h"
+#include "uncrustify.h"
+
+
+using namespace uncrustify;
+
+
+static Chunk *find_end_brace(Chunk *pc)
+{
+   while (!chunk_is_opening_brace(pc))
+   {
+      pc = pc->GetNextNcNnl();
+   }
+   return(pc->SkipToMatch());
+}
+
+
+static bool for_needs_rewrite(Chunk *pc, E_Token desired_type)
+{
+   assert(chunk_is_token(pc, CT_FOR));
```

Comment:
better to use 'if' rather than assert and return if the condition is not met, IMO. Likewise in the rest of the file.